### PR TITLE
rb-gettext: support ruby 3.3, fix installation, add missing deps

### PR DESCRIPTION
**Pull request by barracuda156**
#### Description

Fixups and support for Ruby 3.3.

###### Type(s)

- [x] bugfix
- [x] enhancement
- [ ] security fix

###### Tested on

macOS 14.2.1
Xcode 15.2

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [x] checked your Portfile with `port lint --nitpick`?
- [ ] tried existing tests with `sudo port test`?
- [ ] tried a full install with `sudo port -vst install`?
- [ ] tested basic functionality of all binary files?
- [ ] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

### CODE REVIEW

**Comment by reneeotten** [comment]
If you're anyway plan to fix something yourself, just submit a PR and don't open a Trac ticket. There is no point to that...

**Review by reneeotten — CHANGES_REQUESTED** [review]

**Comment by barracuda156** [comment]
@reneeotten Fair enough, I just looked into it and decided it is easier to fix it straightaway.

File: ruby/rb-erubi/Portfile [hunk 1]
```diff
@@ -0,0 +1,19 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+PortGroup           ruby 1.0
+
+ruby.branches       3.3 3.2 3.1 3.0 2.7 2.6 2.5
```
[the remark this answers is not in the document]
Comment (barracuda156) [reply ruby/rb-erubi/Portfile:6]: @reneeotten Awesome: looks like nothing depends on `rb2*-gettext`, so these can be dropped.

**Comment by kimuraw** [comment]
NOTE: port:sup depends rb-gettext (ruby1.8).  
I will remove sup from macports-ports.

File: ruby/rb-erubi/Portfile (hunk 1 above)
[the remark this answers is not in the document]
Comment (kimuraw) [reply ruby/rb-erubi/Portfile]: (scheduled) EOL date of ruby versions:

- 2.7: 2023-03-31
- 3.0: 2024-03-31
- 3.1: 2025-03-31
- 3.2: 2026-03-31

https://www.ruby-lang.org/en/downloads/branches/

[the remark this answers is not in the document]
Comment (reneeotten) [reply ruby/rb-erubi/Portfile]: thanks; so as I said everything below version 3 should be dropped.

**Comment by barracuda156** [comment]
@kimuraw Finally, CI pass.

**Review by judaew — APPROVED** [review]
LGTM

**Pull request merged**